### DISCUSSION TITLE
API file upload

### DISCUSSION THREAD

**Discussion by ntatko** [directus/directus#5658] [Archive]
Hello!

I'm uploading a photo from an `<input type="file" accept="image/*" onChange={handleChange} />` React element, and I am attempting to load that file into directus at the `/file` endpoint using fetch ``await fetch(`${process.env.API_URL}/files`, { method: "POST", headers: {'Content-Type': 'image/jpeg'}, body: file})``, but I'm getting back the following error, and I don't see them documented anywhere:

```json
{"errors":[
  {"message":"\"storage\" is required","extensions":{"code":"FAILED_VALIDATION","field":"storage","type":"required"}},
  {"message":"\"filename_download\" is required","extensions":{"code":"FAILED_VALIDATION","field":"filename_download","type":"required"}}
]}
```

**Comment by rijkvanzanten** [comment]
Make sure to send up the file as a multipart form data binary string: https://docs.directus.io/reference/api/system/files/#upload-a-file 👍🏻
